snuba/migrations/runner.py

```diff
@@ -279,6 +291,10 @@ def run_migration(
 
         migration_group, migration_id = migration_key
 
+        check_for_inactive_replicas(
+            get_clickhouse_clusters_for_migration_group(migration_group)
+        )
+
```

Comment:
I'm pretty sure you only need this check in `_run_migration_impl`. Right now you are repeating the same check three times, which will make the codebase more brittle over time. Try to find a place where this can be specified once.

Reply:
I could do this, but I will still need to add this check to `_reverse_migration_impl` as well. It will also end up sending more queries to clickhouse as it will query all local nodes in a cluster for each migration. If you're okay with that, I'll move it.

Reply:
I moved it since it does simplify the code